flask_rebar/swagger_generation/swagger_generator_v3.py

```diff
@@ -0,0 +1,322 @@
+                        sw.in_: sw.path,
+                        sw.style: sw.simple,
+                        sw.schema: {
+                            sw.type_: self.flask_converters_to_swagger_types[
+                                path_arg.type
+                            ]
+                        },
+                    }
+                    for path_arg in path_args
+                ]
+
+                # We have to check for an ugly case here. If different Flask
+                # paths that map to the same Swagger path use different URL
+                # converters for the same parameter, we have a problem. Let's
+                # just throw an error in this case.
+                if sw.parameters in path_definition:
+                    verify_parameters_are_the_same(
+                        path_definition[sw.parameters], path_params
+                    )
+
+                path_definition[sw.parameters] = path_params
+
+            for method, d in methods.items():
+                responses_definition = {
+                    sw.default: self._get_response_definition(
+                        self.default_response_schema
+                    )
+                }
+
+                if d.marshal_schema:
+                    for status_code, schema in d.marshal_schema.items():
+                        if schema is not None:
+                            response_definition = self._get_response_definition(schema)
+                        else:
+                            response_definition = {sw.description: "No response body."}
+
+                        responses_definition[str(status_code)] = response_definition
+
+                parameters_definition = []
+
+                if d.query_string_schema:
+                    parameters_definition.extend(
+                        self._convert_schema_to_list_of_parameters(
+                            schema=d.query_string_schema,
+                            converter=self._query_string_converter,
+                            in_=sw.query,
+                        )
+                    )
+
+                if d.headers_schema is USE_DEFAULT:
+                    headers_schema = default_headers_schema
+                else:
+                    headers_schema = d.headers_schema
+
+                if headers_schema:
+                    parameters_definition.extend(
+                        self._convert_schema_to_list_of_parameters(
+                            schema=headers_schema,
+                            converter=self._headers_converter,
+                            in_=sw.header,
+                        )
+                    )
+
+                request_body = None
+
+                if d.request_body_schema:
+                    schema = d.request_body_schema
+
+                    request_body = {
+                        sw.required: True,
+                        sw.content: {
+                            "application/json": {
```

Comment:
Should this be configurable via `consumes`? Or are we just saying JSON only?

Reply:
I think the best thing to do is add proper content negotiation to flask-rebar (@juliusiv had a PR about this a while ago: https://github.com/plangrid/flask-rebar/pull/51)
We could also allow "consumes" and "produces" to be set, and use this value accordingly, but that will be shoving the Swagger v2 interface into this one.

Reply:
I was hoping to solve that problem after this PR, since this one got hefty enough.

Do you think we need to suppose the whole "application/json+plangrid" thing before releasing this?

Reply:
Personally I'm OK with leaving this hard-coded as-is and opening an issue to add proper content-negotiation (which, if it's required for something to move from v2 to v3, should be a great motivator ;))
Alongside that, I'd also suggest we remove those extra params noted above.

Reply:
Per follow-on conversations with Joe, we can punt this to later if it becomes needed